docs/how-tos/build-a-plugin.md

```diff
@@ -0,0 +1,159 @@
+
+```python
+"""MyDataReader implementation for the DataConverter to convert mydata to NeXus."""
+from typing import Tuple, Any
+
+from pynxtools.dataconverter.readers.base.reader import BaseReader
+
+class MyDataReader(BaseReader):
+    """MyDataReader implementation for the DataConverter to convert mydata to NeXus."""
+
+    def read(
+        self,
+        template: dict = None,
+        file_paths: Tuple[str] = None,
+        objects: Tuple[Any] = None
+    ) -> dict:
+        """Reads data from given file and returns a filled template dictionary"""
+        # Fill the template
+        for path in file_paths:
+            print(path)
+
+        template["/entry/instrument/scan"] = raw_scan_data
+
+        return template
+
+
+# This has to be set to allow the convert script to use this reader. Set it to "MyDataReader".
+READER = MyDataReader
+
+
+
+```
+### The reader template dictionary
+
+The read function takes a [`Template`](https://github.com/FAIRmat-NFDI/pynxtools/blob/master/src/pynxtools/dataconverter/template.py) dictionary, which is used to map from the measurement (meta)data to the concepts defined in the NeXus application definition. The template contains keys that match the concepts in the provided NXDL file (you can get an empty template by using `dataconverter generate-template`).
+
+The returned template dictionary should contain keys that exist in the template as defined below. The values of these keys have to be data objects to populate the output NeXus file.
+They can be lists, numpy arrays, numpy bytes, numpy floats, numpy ints, ... . Practically you can pass any value that can be handled by the `h5py` package.
+
+Example for a template entry:
+
+```json
+{
+  "/entry/instrument/source/type": "None"
+}
+```
+
+For a given NXDL schema, you can generate an empty template with the command
+```console
+user@box:~$ dataconverter generate-template` --nxdl NXmynxdl
+```
+
+#### Naming of groups
+In case the NXDL does not define a `name` for the group the requested data belongs to, the template dictionary will list it as `/NAME_IN_NXDL[name_in_output_nexus]`
+You can choose any name you prefer instead of the suggested name (see [here](../learn/nexus-rules.md) for the naming conventions). This allows the reader function to repeat groups defined in the NXDL to be outputted to the NeXus file.
+
+```json
+{
+  "/ENTRY[my_entry]/INSTRUMENT[my_instrument]/SOURCE[my_source]/type": "None"
+}
+```
+
+#### Attributes
+For attributes defined in the NXDL, the reader template dictionary will have the assosciated key with a "@" prefix to the attributes name at the end of the path:
+
+```json
+{
+  "/entry/instrument/source/@attribute": "None"
+}
+```
+
+#### Units
+If there is a field defined in the NXDL, the converter expects a filled in /data/@units entry in the template dictionary corresponding to the right /data field unless it is specified as NX_UNITLESS in the NXDL. Otherwise, you will get an exception.
```

Comment:
changed. Note that I did not use the suggestion because there was something messing up when I tried, but rather I changed it locally. Same for all other suggestions.